optuna/pruners/hyperband.py

```diff
@@ -70,20 +79,36 @@ class HyperbandPruner(BasePruner):
     """
 
     def __init__(
-        self, min_resource=1, reduction_factor=3, n_brackets=4, min_early_stopping_rate_low=0
-    ):
-        # type: (int, int, int, int) -> None
-
+        self,
+        max_resource: int,
+        min_resource: int = 1,
+        reduction_factor: int = 3,
+        n_brackets: Optional[int] = None,
+        min_early_stopping_rate_low: int = 0,
+    ) -> None:
+        self._max_resource = max_resource
         self._pruners = []  # type: List[SuccessiveHalvingPruner]
         self._reduction_factor = reduction_factor
         self._resource_budget = 0
-        self._n_brackets = n_brackets
+
+        if n_brackets is None:
+            self._n_brackets = numpy.ceil(numpy.log2(max_resource) / numpy.log2(reduction_factor))
```

Comment:
I think both of `max_resource` and `reduction_factor` are scalars. Is it possible to replace `numpy`'s `ceil` and `log2` with those in the `math` module?

```suggestion
            self._n_brackets = math.ceil(math.log2(max_resource) / math.log2(reduction_factor))
```

Reply:
And if possible, could you leave the reference of this logic as a code comment?

Reply:
Yes, I agree with you. We can use the `math` module instead of `numpy` module. I will also leave the reference of this logic.